workers/tulip/__init__.py

```diff
@@ -0,0 +1,132 @@
+                    field_name = algorithm + "_" + field
+
+                    properties[field_name] = {
+                        'type': 'string',
+                        'title': field_name,
+                        'default': str(default_params[field])
+                    }
+
+                    dependencies[field_name] = ['algorithm']
+
+                    options[field_name] = {'dependencies': {'algorithm': algorithm}}
+
+        return {
+            'schema': {
+                'type': 'object',
+                'properties': properties,
+                "dependencies": dependencies
+            },
+            "options": {
+                "fields": options
+            }
+        }
+
+    def process(self, project_name: str, meta: OveMeta, options: Dict):
+        logging.info("Copying %s/%s into the temp place ...", project_name, meta.name)
+
+        if not options:
+            options = {}
+
+        with TemporaryDirectory() as folder:
+
+            with TemporaryDirectory() as folder2:
+                network_file = os.path.join(folder2, meta.filename)
+                open(network_file, 'a').close()
+
+                self._file_controller.download_asset(project_name=project_name, asset_name=meta.name,
+                                                     filename=meta.file_location, down_filename=network_file)
+
+                algorithm = options.get('algorithm')
+                if not algorithm:
+                    algorithm = "FM^3 (OGDF)"
+
+                params = tlp.getDefaultPluginParameters(algorithm)
+                for param in params:
+                    params[param] = self.convert_param(options.get(param, params[param]))
+
+                logging.info("Performing layout using algorithm %s and options %s ...", algorithm, params)
+                graph = tlp.loadGraph(network_file)
+                graph.applyLayoutAlgorithm(algorithm, params)
+
+                result_name = options.get('result_name')
+                if not result_name:
+                    result_name = meta.name.split('.')[0] + '.gml'
+                tlp.saveGraph(graph, os.path.join(folder, result_name))
+
+                with open(os.path.join(folder, result_name + ".options"), 'w') as fp:
+                    json.dump(options, fp)
+
+                self._file_controller.upload_asset_folder(project_name=project_name, meta=meta, upload_folder=folder,
+                                                          worker_name=self.name)
+
+        base_name = os.path.splitext(os.path.basename(meta.filename))[0]
+        meta.index_file = os.path.join(meta.worker_root + self.name, base_name, result_name)
+
+        self._file_controller.set_asset_meta(project_name, meta.name, meta)
+        logging.info("Finished generating %s/%s into the storage ...", project_name, meta.name)
+
+    @staticmethod
+    def convert_param(param):
+        if not isinstance(param, str):
+            return param
+        elif str.lower(param) == 'true':
```

Comment:
for str to bool conversion there is a function called to_bool in the commons.util module

Reply:
That function throws an exception if it passed something that is not a string representation of a bool: I just want to defensively ensure that *if* a string is a representation of a boolean then it is converted to an actual boolean.

Reply:
I realized that the to_bool function does not do the conversion you require